bin/setup-bench.mjs

```diff
@@ -0,0 +1,155 @@
+
+*/
+
+const experimentBranchName =
+  process.env['EXPERIMENT_BRANCH_NAME'] || (await $`git rev-parse --abbrev-ref HEAD`).stdout.trim();
+const controlBranchName = process.env['CONTROL_BRANCH_NAME'] || 'main';
+const markers =
+  process.env['MARKERS'] ||
+  'navigationStart,renderStart,renderEnd,glimmer-render-1000-rows-start,glimmer-render-1000-rows-finished';
+const fidelity = process.env['FIDELITY'] || '20';
+const throttleRate = process.env['THROTTLE'] || '4';
+const FORK_NAME = process.env['FORK_NAME'] || '';
+
+const tempDir = os.tmpdir();
+
+const CONTROL_DIR = join(tempDir, 'control');
+const EXPERIMENT_DIR = join(tempDir, 'experiment');
+
+const CONTROL_BENCH_DIR = join(CONTROL_DIR, 'benchmark', 'benchmarks', 'krausest');
+const EXPERIMENT_BENCH_DIR = join(EXPERIMENT_DIR, 'benchmark', 'benchmarks', 'krausest');
+
+const pwdRaw = await $`pwd`;
+const pwd = pwdRaw.toString().trim();
+
+// we use benchmark from current commit, very useful if we need to tweak it
+const benchmarkFolder = 'benchmark';
+
+// remove node_modules from benchmark folder, maybe we could figure out better option to distribute bench source
+await $`rm -rf ${join(pwd, benchmarkFolder, 'node_modules')}`;
+await $`rm -rf ${join(pwd, benchmarkFolder, 'benchmarks', 'krausest', 'node_modules')}`;
+
+await $`rm -rf ${CONTROL_DIR}`;
+await $`rm -rf ${EXPERIMENT_DIR}`;
+await $`mkdir ${CONTROL_DIR}`;
+await $`mkdir ${EXPERIMENT_DIR}`;
+
+const isMacOs = os.platform() === 'darwin';
+
+const BENCHMARK_FOLDER = join(pwd, benchmarkFolder);
+
+const rawUpstreamUrl = await $`git ls-remote --get-url upstream`;
+const rawOriginUrl = await $`git ls-remote --get-url origin`;
+let originUrlStr = rawOriginUrl.toString().trim();
+let upstreamUrlStr = rawUpstreamUrl.toString().trim();
+
+if (upstreamUrlStr === 'upstream') {
+  // if we not inside fork, falling back to origin
+  upstreamUrlStr = originUrlStr;
+}
+
+if (FORK_NAME && FORK_NAME !== 'glimmerjs/glimmer-vm') {
+  // if PR from fork, we need to resolve fork's commit
+  originUrlStr = originUrlStr.replace('glimmerjs/glimmer-vm', FORK_NAME);
+}
+
+const CONTROL_PORT = 4020;
+const EXPERIMENT_PORT = 4021;
+const CONTROL_URL = `http://localhost:${CONTROL_PORT}`;
+const EXPERIMENT_URL = `http://localhost:${EXPERIMENT_PORT}`;
+
+// we can't do it in parallel on CI,
+
+// setup experiment
+await within(async () => {
+  await cd(EXPERIMENT_DIR);
+  await $`git clone ${originUrlStr} .`;
+  await $`git checkout ${experimentBranchName}`;
+  await $`rm -rf ./benchmark`;
+  await $`cp -r ${BENCHMARK_FOLDER} ./benchmark`;
+
+  console.info('installing experiment source');
+  await $`pnpm install --no-frozen-lockfile`.quiet();
+  console.info('building experiment source, may take a while');
+  await $`pnpm build`.quiet();
+
+  if (isMacOs) {
+    await $`find ./packages -name 'package.json' -exec sed -i '' 's|"main": "index.ts",|"main": "./dist/prod/index.js","module": "./dist/prod/index.js",|g' {} \\;`;
```

Comment:
note for myself and a later PR, I bet we can do this from part of the script that @chancancode made that allows the linking, where a whole new monorepo is created, but just with the published assets and resolved publishConfigs

Reply:
yep, or we could expose special "build-like-for-bublishing" command (and use it inside folders) to avoid this magic.

Reply:
yeah, I think that's what I'll make (already implemented for the linking) and share it with this code